automation-tests/test/scratchOrg.ts

```diff
@@ -0,0 +1,328 @@
+
+          // Set the current scratch org.
+          await this.setDefaultOrg(workbench, this.scratchOrgAliasName);
+
+          utilities.log(`${this.testSuiteSuffixName} - found one: ${this.scratchOrgAliasName}`);
+          utilities.log(`${this.testSuiteSuffixName} - ...finished createDefaultScratchOrg()`);
+          utilities.log('');
+          return;
+        }
+      }
+    }
+
+    const definitionFile = path.join(this.projectFolderPath, this.tempProjectName, 'config', 'project-scratch-def.json');
+
+    // Org alias format: TempScratchOrg_yyyy_mm_dd_username_ticks_testSuiteSuffixName
+    const currentDate = new Date();
+    const ticks = currentDate.getTime();
+    const day = ('0' + currentDate.getDate()).slice(-2);
+    const month = ('0' + (currentDate.getMonth() + 1)).slice(-2);
+    const year = currentDate.getFullYear();
+    this.scratchOrgAliasName = `TempScratchOrg_${year}_${month}_${day}_${userName}_${ticks}_${this.testSuiteSuffixName}`;
+    utilities.log(`${this.testSuiteSuffixName} - temporary scratch org name is ${this.scratchOrgAliasName}...`);
+
+    const startDate = Date.now();
+    const durationDays = 1;
+
+    utilities.log(`${this.testSuiteSuffixName} - calling sfdx force:org:create...`);
+    const sfdxForceOrgCreateResult = await exec(`sfdx force:org:create -f ${definitionFile} --setalias ${this.scratchOrgAliasName} --durationdays ${durationDays} --setdefaultusername --json --loglevel fatal`);
+    const json = this.removedEscapedCharacters(sfdxForceOrgCreateResult.stdout);
+    const result = JSON.parse(json).result;
+
+    const endDate = Date.now();
+    const time = endDate - startDate;
+    utilities.log(`Creating ${this.scratchOrgAliasName} took ${time} ticks (${time/1000.0} seconds)`);
+
+    expect(result.authFields).not.toBeUndefined();
+    expect(result.authFields.accessToken).not.toBeUndefined();
+    expect(result.orgId).not.toBeUndefined();
+    expect(result.scratchOrgInfo.SignupEmail).toEqual(EnvironmentSettings.getInstance().devHubUserName);
+
+    // Run SFDX: Set a Default Org
+    utilities.log(`${this.testSuiteSuffixName} - selecting SFDX: Set a Default Org...`);
+    const inputBox = await utilities.executeQuickPick(workbench, 'SFDX: Set a Default Org');
+
+    // Wait for the quick pick list to appear.
+    await utilities.pause(1);
+
+    // Select this.scratchOrgAliasName from the list.
+    let scratchOrgQuickPickItemWasFound = false;
+    const quickPicks = await inputBox.getQuickPicks();
+    for (const quickPick of quickPicks) {
+      const label = await quickPick.getLabel();
+      // Find the org that was created.
+      if (label.includes(this.scratchOrgAliasName)) {
+        await quickPick.select();
+        await utilities.pause(3);
+        scratchOrgQuickPickItemWasFound = true;
+        break;
+      }
+    }
+    expect(scratchOrgQuickPickItemWasFound).toBe(true);
+    // Warning! This only works if the item (the scratch org) is visible.
```

Comment:
Since this is part of test setup and not the actual code under test, this seems like it might be a good opportunity to use the CLI for consistent test expectations, and that we should create a spec specifically for org creation to test this flow.

Reply:
Great point, I'll go and change calls to `expect()` to actual throws in scratchOrg.ts and utilities.ts

Reply:
done